modules/ocf/templates/apt/kubernetes-pin.pref.erb

```diff
@@ -0,0 +1,9 @@
+# This file is managed by Puppet. DO NOT EDIT.
+# Pins each package in $kube_packages to the hiera-specified version,
+# so it won't get upgraded automatically by apt.
+<% @kube_packages.each do |pkg| -%>
+Package: <%= pkg %>
+Pin: version <%= scope().call_function('hiera', ['kubernetes::kubernetes_package_version']) %>
```

Comment:
Instead of grabbing from hiera like this, I think you should do it like in https://github.com/ocf/puppet/blob/master/modules/ocf_kubernetes/manifests/master.pp#L38 and https://github.com/ocf/puppet/blob/master/modules/ocf_kubernetes/templates/static-tokens.csv.erb